### ISSUE TITLE
iOS 15: New built-in 2FA feature of iCloud Keychain not supported

### ISSUE THREAD

**Issue by AliSoftware** [wordpress-mobile/WordPress-iOS#17203]
### Expected behavior

When login into the WordPress app with my iCloud Keychain, while I saved my 2FA code into my iOS Keychain entry for WordPress ([new feature of iOS 15](https://www.macstories.net/stories/ios-and-ipados-15-the-macstories-review/19/#passwords)), the 2FA code should be auto-filled.

### Actual behavior

After logging out, when I select my account in the picker to log back in and get to the 2FA screen, I don't have the option to take advantage of the fact that my 2FA code is available in my iOS Keychain.

### Steps to reproduce the behavior

- Open your iPhone Settings > Passwords
- Select the password entry for your wordpress.com account (assuming you have store your WP password in iCloud Keychain in the past. If not, do so first)
- Tap on "Set Up Verification Code", then scan QR code
- Go to your wordpress.com account settings in your browser, disable 2FA, then re-enable it, to restart the 2FA enrolment process. Once you are shown the QR code in your browser, scan it with your iPhone
- Finish the 2FA enrolment process

At that stage, your 2FA code for your WP account should be stored in your iCloud Keychain item for WP. Then

 - Open the WP iOS app
 - Log out
 - Select your account in the account picker to log back in
 - When you get to the 2FA screen, notice that iOS doesn't auto-fill the 2FA code you just registered in your iCloud Keychain

### Probable Solution

According to the [Apple Docs](https://developer.apple.com/documentation/security/password_autofill/enabling_password_autofill_on_a_text_input_view), all we should have to do to support this is to set the input type of the field to `.oneTimeCode`. At least that's the theory 😛

##### Tested on [device], iOS [version], WPiOS [version]

iPhone 11 Pro, iOS 15.0, WPiOS 18.3.0.0

**Comment by mokagio** [comment]
Bumped to 18.5 because this hasn't been picked up yet and I'll be starting the 18.4 code freeze today.

We currently [don't _require_ issues to have a milestone](https://github.com/Automattic/peril-settings/blob/master/peril-settings.json#L13). I was tempted to remove it, but I think this is such a small (hopefully) thing to tackle and one that would make the users' experience much better. It deserves to stay top of mind.

**Comment by mokagio** [comment]
Bumped to 18.6 because this hasn't been picked up yet and I'll be starting the 18.5 code freeze today.

See also rationale from [comment above](https://github.com/wordpress-mobile/WordPress-iOS/issues/17203#issuecomment-933076096).

**Comment by oguzkocer** [comment]
It looks like this has been bumped a few times without any activity, so I am removing the milestone for the issue until there is a plan to work on it for a certain milestone.

**Comment by AliSoftware** [comment]
I just did a quick check of the code today and noticed that [the One Time Code attribute was set](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/develop/WordPressAuthenticator/Signin/Login.storyboard#L433) on the 2FA fields.

I then tested on my device with the latest WPiOS 18.6.0.0 and got iOS suggest me an OTP on the top of my keyboard to use and successfully auto-filled with it when I tapped the suggestion.

Not sure what changed since my last test (maybe it was fixed in Login lib, or I tested it with an early version of iOS 15 while I'm now on 15.1?) but seems it's all fixed now, so I'm gonna close this one.
